nix/buildLeanPackage.nix

```diff
@@ -44,11 +50,19 @@ with builtins; let
     allowSubstitutes = false;
   };
   srcRoot = src;
-  allDeps = lib.foldr (dep: allDeps: allDeps // { ${dep.name} = dep; } // dep.allDeps) {} deps;
+
+  # A flattened list of Lean-module dependencies (`deps`)
+  allExternalDeps = lib.foldr (dep: allExternalDeps: allExternalDeps // { ${dep.name} = dep; } // dep.allExternalDeps) {} deps;
+  # A flattened list of all static library dependencies: this and every dep module's explicitly provided `staticLibDeps`, 
+  # plus every dep module itself: `dep.staticLib`
+  allStaticLibDeps = 
+    lib.unique (lib.flatten (staticLibDeps ++ (map (dep: [dep.staticLib] ++ dep.staticLibDeps or []) (attrValues allExternalDeps))));
+  leanPluginFlags = lib.concatStringsSep " " (map (dep: "--plugin=${dep}/${dep.name}") pluginDeps);
```

Comment:
Is this path missing a `.so`?

Reply:
The `.so` is in the .sharedLib derivation's name. One problem here is that the `.sharedLib` derivation (and `.staticLib`) loses the references to the original mod's name. A more principled fix here would be to have `.sharedLib` and `.staticLib` inherit `mod` and form paths based on `${dep.mod.name}.so` and `lib${dep.mod.name}.a` instead of `${dep.name}`, which we'd ideally keep different from the filename for readability reasons.

Reply:
(and fwiw, I "tested" these changes by making sure my plugin-and-static-lib-using-project compiles)

Reply:
Oh of course, it's in the name! I rather like that, perhaps I should change `staticLib` in the same way.